cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/DatabaseImpl.java

```diff
@@ -1191,4 +1186,43 @@ public DocumentRevision delete(final DocumentRevision rev) throws
         }
     }
 
+    @Override
+    public void createWithHistory(DocumentRevision revision, int revisionsStart, List<String>
+            revisionsIDs) throws DocumentException {
+
+        Map<String, Attachment> attachments = revision.getAttachments();
+        Collections.list(Collections.enumeration(revision.getAttachments().values()));
+
+        InternalDocumentRevision internalRev = new DocumentRevisionBuilder()
+                .setDocId(revision.getId())
+                .setRevId(revision.getRevision())
+                .setBody(revision.getBody())
+                .setAttachments(attachments)
+                .setDeleted(revision.isDeleted())
+                .build();
+
+        Map<String[], Map<String, PreparedAttachment>> preparedAttachments = Collections
+                .singletonMap
+                        (new String[]{revision.getId(), revision.getRevision()}, AttachmentManager
+                                .prepareAttachments(attachmentsDir, attachmentStreamFactory,
+                                        attachments));
+
+        Map<String, Object> attachmentsMetadata = new HashMap<String, Object>();
```

Comment:
Clarified the code commenting around this in daac2ed0e1a399008a9d35f1f6938837275ec0c3.
Will follow with additional testing of existing attachments and more example doc of how to handle attachments.